### DISCUSSION TITLE
Debug using JS console possible?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [uwdata/mosaic#289] [Q&A]
I want to query the current DuckDB (connection) from the JS console for debugging, but I can't determine how to get the result from the promise. Does anyone have a tip on how to do that?

```
→ (async() → {const t = await vg. coordinator () exec("show result_table"); console. log(t.toString()) ;})();
* › Promise {[[PromiseState]l: "pending", [[PromiseResult]l: undefined}
→ vg. coordinator (). exec("show result_table"). then(result → console.log(result.toString()))
+ › Promise {[[PromiseState]]: "pending", [[PromiseResult]]: undefined}
```

**Comment by jheer** [comment]
If you have access to the Coordinator, you can issue queries through `vg.coordinator().exec(...)` (for queries that do not return anything!) or [`vg.coordinator().query(sql, options)`](https://github.com/uwdata/mosaic/blob/main/packages/core/src/Coordinator.js#L74) for queries that return a table.

Or, if you'd rather bypass the Coordinator's internal query manager (and related caching and consolidation), you can get the current database connector using `vg.coordinator().databaseConnector()`. You can then query DuckDB directly using `connector.query(queryObject)`. The queryObject argument has the shape `{ type, sql }`, where `sql` is the SQL query string and `type` is the return type: one of `'exec'` (return nothing), `'arrow'` (Apache Arrow table), or `'json'` (vanilla JS objects). The query method is async, so you'll have to `await` the resulting `Promise`.

**Comment by Robert-M-Muench** [comment]
Using the `query()` doesn't help, still just get a promise:
<img width="584" alt="image" src="https://github.com/uwdata/mosaic/assets/1826391/ea87dc39-b6e8-40ea-8661-02a8c6e11727">

Same for the `databaseConnector()` approach:
<img width="635" alt="image" src="https://github.com/uwdata/mosaic/assets/1826391/b305c671-24df-4709-9d67-76016cfa6f2e">

That's precisely my problem. I don't want to promise, but the SQL result is a string.

**Reply by jheer** [reply]
That's strange. I've had no problem doing this so long as I include `await`, which should cause you to get the Promise result once complete, not the Promise itself. Maybe double check what you're getting after each chained call? Or if all else fails, `await` the result of the first `await` and see what you're getting?

Also: note that if you are using the db connector, you can't pass the SQL string directly. As I described above, you need to provide an object of the form `{ type, sql }`, such as `databaseConnector.query({ type: 'json', sql: 'SELECT * ...' })`.
